tests/functional/testOnboarding.js

```diff
@@ -369,4 +396,366 @@ describe('Onboarding', function() {
+    const dataCollectionScreenEventsExtras = dataCollectionScreenEvents[0].extra;
+    assert.equal(dataScreenTelemetryId, dataCollectionScreenEventsExtras.screen);
+
+    await advanceToSlide(1)
+
+    //Verify that getMorePrivacyScreen event is recorded
+    const getMorePrivacyScreenEvents = await vpn.gleanTestGetValue("impression", "getMorePrivacyScreen", "main");
+    assert.equal(getMorePrivacyScreenEvents.length, 1);
+    const getMorePrivacyScreenEventsExtras = getMorePrivacyScreenEvents[0].extra;
+    assert.equal(privacyScreenTelemetryId, getMorePrivacyScreenEventsExtras.screen);
+
+    await advanceToSlide(2)
+
+    //Verify that installOn5DevicesScreen event is recorded
+    const installOn5DevicesScreenEvents = await vpn.gleanTestGetValue("impression", "installOn5DevicesScreen", "main");
+    assert.equal(installOn5DevicesScreenEvents.length, 1);
+    const installOn5DevicesScreenEventsExtras = installOn5DevicesScreenEvents[0].extra;
+    assert.equal(devicesScreenTelemetryId, installOn5DevicesScreenEventsExtras.screen);
+
+    await advanceToSlide(3)
+
+    //Verify that connectOnStartupScreen event is recorded
+    const connectOnStartupScreenEvents = await vpn.gleanTestGetValue("impression", "connectOnStartupScreen", "main");
+    assert.equal(connectOnStartupScreenEvents.length, 1);
+    const connectOnStartupScreenEventsExtras = connectOnStartupScreenEvents[0].extra;
+    assert.equal(startScreenTelemetryId, connectOnStartupScreenEventsExtras.screen);
+  });
+
+  it('Progress bar button events are recorded', async () => {
+    await vpn.gleanTestReset();
+
+    await vpn.waitForQuery(queries.screenOnboarding.DATA_SLIDE.visible());
+
+    //Ensure telemetry is not generated by clicking on the current step
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.STEP_PROG_BAR_DATA_BUTTON.visible());
+    let dataUseSelectedEvents = await vpn.gleanTestGetValue("interaction", "dataUseSelected", "main");
+    assert.equal(dataUseSelectedEvents.length, 0);
+
+
+    //Advance to privacy slide
+    await advanceToSlide(1)
+
+    //Ensure telemetry is not generated by clicking on the current step
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.STEP_PROG_BAR_PRIVACY_BUTTON.visible());
+    let morePrivacySelectedEvents = await vpn.gleanTestGetValue("interaction", "morePrivacySelected", "main");
+    assert.equal(morePrivacySelectedEvents.length, 0);
+
+    //Ensure dataUseSelected event is recorded from the privacy screen
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.STEP_PROG_BAR_DATA_BUTTON.visible());
+    dataUseSelectedEvents = await vpn.gleanTestGetValue("interaction", "dataUseSelected", "main");
+    assert.equal(dataUseSelectedEvents.length, 1);
+    let dataUseSelectedEventsExtras = dataUseSelectedEvents[0].extra;
+    assert.equal(privacyScreenTelemetryId, dataUseSelectedEventsExtras.screen);
+
+
+    //Advance to devices slide
+    await advanceToSlide(2)
+    await vpn.gleanTestReset();
```

Comment:
This applies to all calls of this function.

Reply:
laziness so I didn't have to count aggregate number of calls made to events (since we are emitting the same event multiple times in this test). will fix!